tools/gulptasks/dist-upload.js

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) Highsoft AS
+ */
+
+/* eslint func-style: 0, no-console: 0, max-len: 0 */
+const gulp = require('gulp');
+const glob = require('glob');
+const build = require('./lib/build');
+const { uploadFiles, uploadProductPackage } = require('./lib/uploadS3');
+
+
+const DIST_DIR = 'build/dist';
+
+/**
+ * Upload a distribution version for all products.
+ *
+ * @param {string} productName from build properties, e.g 'highstock'.
+ * @param {object} productProps containing product name and version (build.properties).
+ * @return {Promise<any> | Promise | Promise} Promise to keep
+ */
+function uploadPackage(productName, productProps) {
+    const { name, version, cdnpath } = productProps;
+    return uploadProductPackage(productName, cdnpath, name, version);
+}
+
+/**
+ * Upload a distribution version for all products.
+ *
+ * @param {Array<string>} products (product names) to initiate an upload for.
+ * @return {Promise<any> | Promise | Promise} Promise to keep
```

Comment:
Here you could do the same approach and add a `.then(() => undefined)` to the returning `Promise.all` to cast it to a simple `Promise<void>`. Gulp does not care about resolve results - only reject errors are handled (for now).

Reply:
Changed return type to Promise<*> for now.